helix-term/src/application.rs

```diff
@@ -245,6 +251,38 @@ impl Application {
         }
     }
 
+    pub fn refresh_config(&mut self) {
```

Comment:
This doesn't need to be a separate event processed on the Application level, both `application.config` and `editor.config` point to the same `ArcSwap<Arc<T>>` so it's enough to add a `editor.reload_config` method that loads and updates the config via `config.store`.